File: libs/document-processor/src/__tests__/models/recency-test-channel.ts
import { z } from 'zod';
import { withTypeBlueId } from '@blue-labs/language';

import { channelContractBaseSchema } from '../../model/shared/index.js';

export const recencyTestChannelSchema = withTypeBlueId('RecencyTestChannel')(
  channelContractBaseSchema.extend({
    minDelta: z.number().optional(),
  }),
);

export type RecencyTestChannel = z.infer<typeof recencyTestChannelSchema>;

File: references/java-sdk/src/test/java/blue/language/processor/model/RecencyTestChannel.java
package blue.language.processor.model;

import blue.language.model.TypeBlueId;
import blue.language.processor.model.ChannelContract;

@TypeBlueId("RecencyTestChannel")
public class RecencyTestChannel extends ChannelContract {

    private Integer minDelta;

    public Integer getMinDelta() {
        return minDelta;
    }

    public void setMinDelta(Integer minDelta) {
        this.minDelta = minDelta;
    }
}
